### PR TITLE
Upgrading to v11 of the Azure SDK for Go

### PR DESCRIPTION
**Pull request by tombuildsstuff**
Upgrading to v11 of the Azure SDK for Go

- [x] Vendoring `v11.0.0-beta` of the Azure SDK
- [x] State Migrations for:
  - [x] Container Registry
      - [x] `Basic` SKU -> `Classic`
      - [x] State Migration Storage Account Name & Key -> Storage Account ID
  - [x] Key Vault  
    - [x] `certificate_permissions` All -> every option
    - [x] `key_permissions` All -> every option
    - [x] `secret_permissions` All -> every option
  - [x] ServiceBus Namespace
    - [x] Remove `capacity` if not a `premium` sku
- [x] Pending issues:
  - [x] ~[Traffic Manager API version used in the SDK isn't available](https://github.com/Azure/azure-sdk-for-go/issues/794)~ fixed
  - [x] ~[Container Registry API version used in the SDK isn't available](https://github.com/Azure/azure-sdk-for-go/issues/795)~ fixed
 - [ ] Ensure all breaking changes are updated in the Documentation

### Changes and Deprecation Notices:

  - `azurerm_automation_account` - the SKU `Free` has been replaced with `Basic`.
  - `azurerm_container_registry` - Azure has updated the SKU from `Basic` to `Classic`, with new `Basic`, `Standard` and `Premium` SKU's introduced.
  - `azurerm_container_registry` - the `storage_account` block is now `storage_account_id` and is only required for `Classic` SKU's
  - `azurerm_key_vault` - `certificate_permissions`, `key_permissions` and `secret_permissions` have all had the `All` option removed by Azure. Each permission now needs to be specified manually.
  - `azurerm_servicebus_namespace` - The `capacity` field can only be set for `Premium` SKU's
  - `azurerm_servicebus_queue` - The `enable_batched_operations` and `support_ordering` fields have been deprecated by Azure.
  - `azurerm_servicebus_subscription` - The `dead_lettering_on_filter_evaluation_exceptions` has been removed by Azure.
  - `azurerm_servicebus_topic` - The `enable_filtering_messages_before_publishing` field has been removed by Azure.

### CODE REVIEW

**Comment by tombuildsstuff** [comment]
Taking a look into this - there's enough breaking changes that this is going to take some time - as such I'm going to park this PR for the moment.

**Comment by tombuildsstuff** [comment]
I'm just kicking off the full test suite now - but I need to do a ton of checking of the state migrations next

**Comment by tombuildsstuff** [comment]
It appears the Traffic Manager SDK is broken as the API version used isn't rolled out in Azure at the moment - opened an issue on the SDK repository: https://github.com/Azure/azure-sdk-for-go/issues/794

**Comment by tombuildsstuff** [comment]
Container Registry's broken too: https://github.com/Azure/azure-sdk-for-go/issues/795

**Comment by tombuildsstuff** [comment]
This PR will allow the following to be fixed: #272 #339

File: azurerm/resource_arm_servicebus_topic.go [hunk 1]
```diff
@@ -193,41 +191,40 @@ func resourceArmServiceBusTopicRead(d *schema.ResourceData, meta interface{}) er
...
+
+		// if the topic is in a premium namespace and partitioning is enabled then the
+		// max size returned by the API will be 16 times greater than the value set
+		if *props.EnablePartitioning {
+			namespace, err := meta.(*ArmClient).serviceBusNamespacesClient.Get(resGroup, namespaceName)
+			if err != nil {
+				return err
+			}
+
+			if namespace.Sku.Name != servicebus.Premium {
```
Comment (mbfrahry) [inline azurerm/resource_arm_servicebus_topic.go]: I might be reading this wrong but I just want to confirm based on your comment that premium has max size * 16

**Review by mbfrahry — APPROVED** [review]
I have a question about a comment and then the following code snippet. Other than that, this looks really good. Nice job Tom

File: azurerm/resource_arm_servicebus_topic.go (hunk 1 above)
Comment (tombuildsstuff) [reply azurerm/resource_arm_servicebus_topic.go]: this has been around for a while - so it hasn't changed AFAIK

**Comment by vancluever** [comment]
:tada: great job @tombuildsstuff!!!

**Pull request merged**